pkg/storegateway/gateway_ring.go

```diff
@@ -0,0 +1,93 @@
+package storegateway
+
+import (
+	"flag"
+	"os"
+	"time"
+
+	"github.com/go-kit/kit/log/level"
+
+	"github.com/cortexproject/cortex/pkg/ring"
+	"github.com/cortexproject/cortex/pkg/ring/kv"
+	"github.com/cortexproject/cortex/pkg/util"
+	"github.com/cortexproject/cortex/pkg/util/flagext"
+)
+
+// RingConfig masks the ring lifecycler config which contains
+// many options not really required by the store gateways ring. This config
+// is used to strip down the config to the minimum, and avoid confusion
+// to the user.
+type RingConfig struct {
```

Comment:
Would it be possible to reduce repetition with these RingConfig structures too at some point? We have 4 of them now 😞

Reply:
Yes. There're some slightly differences (ie. replication factor and tokens) but we may workaround it in a refactoring to avoid duplication.